src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,447 @@
+        :avocado: tags=control,config_generate_entries,net_class
+        """
+        # Get necessary storage and network info.
+        self.prepare_expected_data()
+
+        # Get ib_count threshold.
+        ib_count = 0
+        for interface in self.interface_set:
+            if interface[:2] == "ib":
+                ib_count += 1
+        self.log.info("ib_count = {}".format(ib_count))
+
+        dmg = self.get_dmg_command()
+        errors = []
+
+        # Call dmg config generate --num-engines=<1 to ib_count>
+        # --net-class=infiniband. Should pass.
+        for num_engines in range(1, ib_count + 1):
+            try:
+                # dmg config generate should pass.
+                generated_config = dmg.config_generate(
+                    access_points="wolf-a", num_engines=num_engines,
+                    net_class="infiniband")
+
+                for engine in generated_config["engines"]:
+                    fabric_iface = engine["fabric_iface"]
+                    provider = engine["provider"]
+                    # Verify fabric_iface field, e.g., ib0 by checking the
+                    # dictionary keys.
+                    if not self.interface_to_providers[fabric_iface]:
+                        errors.append(
+                            "Unexpected fabric_iface! {}".format(fabric_iface))
+                    elif provider not in \
+                        self.interface_to_providers[fabric_iface]:
+                        # Now check the provider field, e.g., ofi+sockets by
+                        # checking the corresponding list in the dictionary.
+                        msg = "Unexpected provider in fabric_iface! provider ="\
+                            " {}; fabric_iface = {}".format(
+                                provider, fabric_iface)
+                        errors.append(msg)
+            except CommandFailure:
+                msg = "config generate failed with --net-class=infiniband "\
+                      "--num-engines = {}!".format(num_engines)
+                errors.append(msg)
+
+        # Call dmg config generate --num-engines=<ib_count + 1>
+        # --net-class=infiniband. Too many engines. Should fail.
+        try:
+            dmg.config_generate(
+                access_points="wolf-a", num_engines=ib_count + 1,
+                net_class="infiniband")
+            msg = "config generate succeeded with --net-class=infiniband "\
+                  "num_engines = {}!".format(ib_count + 1)
+            errors.append(msg)
+        except CommandFailure:
+            msg = "config generate failed with --net-class=infiniband "\
+                  "--num-engines = {} as expected.".format(ib_count + 1)
+            self.log.info(msg)
+
+        # Get eth_count threshold.
+        eth_count = 0
+        for interface in self.interface_set:
+            if interface[:3] == "eth":
+                eth_count += 1
+        self.log.info("eth_count = {}".format(eth_count))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions

Reply:
This should be fixed:
```
        self.log.info("eth_count = %s", eth_count)
```